packages/vite-plugin-angular/src/lib/angular-vite-plugin.ts

```diff
@@ -888,3 +895,28 @@ export function createFileEmitter(
     return { content, dependencies: [], errors, warnings, hmrUpdateCode };
   };
 }
+
+function getDiagnosticsForSourceFile(
+  sourceFile: ts.SourceFile,
+  disableTypeChecking: boolean,
+  program: ts.BuilderProgram,
+  angularCompiler?: NgtscProgram['compiler']
+) {
+  const syntacticDiagnostics = program.getSyntacticDiagnostics(sourceFile);
+
+  if (disableTypeChecking) {
+    // Syntax errors are cheap to compute and the app will not run if there are any
+    // So always show these types of errors regardless if type checking is disabled
+    return syntacticDiagnostics;
```

Comment:
This matches the behaviour of the angular cli where disabling typechecking only disables semantic diagnostics:
https://github.com/angular/angular-cli/blob/9b57ff0552736e6fef44c0ff7e6535f9ed832c6b/packages/angular/build/src/tools/esbuild/angular/compiler-plugin.ts#L378C90-L378C98